src/WordPress/Blueprints/Runner/Step/CpStepRunner.php

```diff
@@ -2,18 +2,53 @@
 
 namespace WordPress\Blueprints\Runner\Step;
 
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+use WordPress\Blueprints\BlueprintException;
 use WordPress\Blueprints\Model\DataClass\CpStep;
 
+class CpStepRunner extends BaseStepRunner
+{
+    /**
+     * @param CpStep $input
+     */
+    public function run(CpStep $input)
+    {
+        $resolvedFromPath = $this->getRuntime()->resolvePath($input->fromPath);
+        $resolvedToPath = $this->getRuntime()->resolvePath($input->toPath);
 
-class CpStepRunner extends BaseStepRunner {
-	/**
-	 * @param CpStep $input
-	 */
-	function run( CpStep $input ) {
-		// @TODO: Treat the input paths as relative path to the document root (unless it's absolute)
-		$success = copy( $input->fromPath, $input->toPath );
-		if ( ! $success ) {
-			throw new \Exception( "Failed to copy file from {$input->fromPath} to {$input->toPath}" );
-		}
-	}
+        $fileSystem = new Filesystem();
+
+        try {
+            // TODO affirm it should overwrite newer files
+            // Filesystem's copy requires flag `overwriteNewerFiles` set to `true`
+
+            // TODO affirm it should copy file to directory
+            // copy throws an exception when $resolvedToPath is a directory - linux would handle this:
+            // When the program has one or more arguments of path names of files and following those an argument
+            // of a path to a directory, then the program copies each source file to the destination directory,
+            // creating any files not already existing.
+
+            // TODO affirm it should copy all files from one directory to another (also, ponder recursivity)
+            // copy throws an exception when $resolvedToPath or $resolvedFromPath are directories - linux would
+            // handle this:
+            // When the program's arguments are the path names to two directories,
+            // cp copies all files in the source directory to the destination directory,
+            // creating any files or directories needed. This mode of operation requires an additional option flag,
+            // typically r, to indicate the recursive copying of directories.
+            // If the destination directory already exists, the source is copied into the destination,
+            // while a new directory is created if the destination does not exist.
+
+            // TODO establish how $input->toPath should be handled
+            // should copy to document root when null or when ""?
+
+            $fileSystem->copy($resolvedFromPath, $resolvedToPath, true);
```

Comment:
I've pondered it for a moment or two. And I think the right way forward here is to implement our own copy method. 

My reasoning: 
- As a user, I'd see a lot of value in using it the way unix specifies it. [source](https://en.wikipedia.org/wiki/Cp_(Unix)#:~:text=Meyering.%5B5%5D-,Operating%20modes,-%5Bedit%5D) 
- Would work more akin to RmStep, handling both files and directories alike.
- We intend to have a limited amount of steps, so it's not like we'll have a ton of custom methods to maintain.
- We'll be less reliant on Filesystem, which we must downgrade to v3.x for PHP 7.0.

Reply:
Pretty much all of the things listed. Since the only thing it does is copy one specific file to a different location. IMHO that's constraining.

Reply:
Let's move this discussion to #58 
Commit with cp-related changes and tests was reverted for this pr.